src/main/java/com/microsoft/graph/requests/BatchRequestBuilder.java

```diff
@@ -0,0 +1,95 @@
+package com.microsoft.graph.requests;
+
+import com.microsoft.graph.CoreConstants;
+import com.microsoft.graph.content.BatchRequestContent;
+import com.microsoft.graph.content.BatchRequestContentCollection;
+import com.microsoft.graph.content.BatchResponseContent;
+import com.microsoft.graph.content.BatchResponseContentCollection;
+import com.microsoft.graph.exceptions.ErrorConstants;
+import com.microsoft.kiota.HttpMethod;
+import com.microsoft.kiota.NativeResponseHandler;
+import com.microsoft.kiota.RequestAdapter;
+import com.microsoft.kiota.RequestInformation;
+import com.microsoft.kiota.serialization.Parsable;
+import com.microsoft.kiota.serialization.ParsableFactory;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import okhttp3.Response;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * A request builder for creating batch requests.
+ */
+public class BatchRequestBuilder {
+    private final RequestAdapter requestAdapter;
+    /**
+     * Instantiates a new BatchRequestBuilder.
+     * @param requestAdapter the adapter to use to build requests.
+     */
+    public BatchRequestBuilder(@Nonnull RequestAdapter requestAdapter) {
+        this.requestAdapter = Objects.requireNonNull(requestAdapter, ErrorConstants.Messages.NULL_PARAMETER + "requestAdapter");
+    }
+    /**
+     * Posts a batch request.
+     * @param requestContent the batch request content.
+     * @param errorMappings the error mappings to use when parsing the response.
+     * @return the batch response content.
+     */
+    @Nonnull
+    public CompletableFuture<BatchResponseContent> postAsync(@Nonnull BatchRequestContent requestContent, @Nullable Map<String, ParsableFactory<? extends Parsable>> errorMappings) {
```

Comment:
Just so that we do not forget, we will want to create a derived instance of this class in the service library that passes the errorMapping collection that exist in the generated code. Check this PR for reference
https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/1794/files